addons/account/models/account_account.py

```diff
@@ -147,38 +147,45 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
         if fname == 'code':
             return self.with_company(self.env.company.root_id).sudo()._field_to_sql(alias, 'code_store', query, flush)
         if fname == 'placeholder_code':
+            # The placeholder_code is defined as the account's code in the first active company to
+            # which the account belongs.
             query.add_join(
-                'JOIN',
+                'LEFT JOIN',
                 'account_first_company',
                 SQL(
-                    """(
-                        SELECT DISTINCT ON (rel.account_account_id)
-                               rel.account_account_id AS account_id,
-                               rel.res_company_id AS company_id,
-                               SPLIT_PART(res_company.parent_path, '/', 1) AS root_company_id,
-                               res_company.name AS company_name
-                          FROM account_account_res_company_rel rel
-                          JOIN res_company
-                            ON res_company.id = rel.res_company_id
-                         WHERE rel.res_company_id IN %(authorized_company_ids)s
-                      ORDER BY rel.account_account_id, company_id
+                    """LATERAL (
```

Comment:
Did you test this? Why are you changing to lateral?
A lateral join is supposed to have a `LIMIT 1` (or an other limit)